src/main/java/org/opensearch/ad/model/AnomalyResult.java

```diff
@@ -223,9 +244,13 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         XContentBuilder xContentBuilder = builder
             .startObject()
             .field(DETECTOR_ID_FIELD, detectorId)
-            .field(DATA_START_TIME_FIELD, dataStartTime.toEpochMilli())
-            .field(DATA_END_TIME_FIELD, dataEndTime.toEpochMilli())
-            .field(CommonName.SCHEMA_VERSION_FIELD, schemaVersion);
+            .field(SCHEMA_VERSION_FIELD, schemaVersion);
+        if (dataStartTime != null) {
+            xContentBuilder.field(DATA_START_TIME_FIELD, dataStartTime.toEpochMilli());
+        }
+        if (dataEndTime != null) {
+            xContentBuilder.field(DATA_END_TIME_FIELD, dataEndTime.toEpochMilli());
+        }
```

Comment:
Can you explain these added checks? What has changed to allow passing null data start/end times?

Reply:
In normal AD result, we always pass data start/end times. In custom result index, we need to write/delete a dummy AD result to verify if user has write permission to the custom result index. Just pass in null start/end time for this dummy anomaly result to make sure it won't be queried by  mistake.

Reply:
I see, can you add a comment in the code about this?